pkg/buildpack/multi_architecture_helper.go

```diff
@@ -0,0 +1,170 @@
+// was copied.
+func PrepareDependencyConfigFile(baseDir, depURI string, target dist.Target, version string, failWhenURI bool) (string, error) {
+	// Only in cases it is a URILocator we may want to copy config files
+	locatorType, err := GetLocatorType(depURI, baseDir, []dist.ModuleInfo{})
+	if err != nil {
+		return "", err
+	}
+
+	if locatorType == URILocator {
+		if failWhenURI {
+			return "", errors.New(fmt.Sprintf("%s is not allowed when creating a composite buildpack, use 'docker://' instead", style.Symbol(depURI)))
+		}
+
+		uri, err := paths.FilePathToURI(depURI, baseDir)
+		if err != nil {
+			return "", errors.Wrapf(err, "making absolute: %s", style.Symbol(depURI))
+		}
+
+		if paths.IsURI(uri) {
+			parsedURL, err := url.Parse(uri)
+			if err != nil {
+				return "", errors.Wrapf(err, "parsing path/uri %s", style.Symbol(uri))
+			}
+
+			if parsedURL.Scheme == "file" {
+				path, err := paths.URIToFilePath(uri)
+				if err != nil {
+					return "", err
+				}
+				if exists, _ := paths.IsDir(path); exists {
+					return CopyConfigFile(path, target, version)
+				}
+			}
+		}
+	}
+	return "", nil
+}
+
+// CopyConfigFile will determine and copy the buildpack.toml file into the corresponding platform folder
+func CopyConfigFile(baseDir string, target dist.Target, version string) (string, error) {
+	if ok, platformRootFolder := PlatformRootFolder(baseDir, target, version); ok {
+		path, err := copy(baseDir, platformRootFolder)
+		if err != nil {
+			return "", err
+		}
+		return path, nil
+	}
+	return "", nil
+}
+
+// PlatformRootFolder calculates the top-most directory that identifies a target in a given <root> folder.
+// Let's define a target with the following format: [os][/arch][/variant]:[name@version], consider the following examples:
+//   - Given a target linux/amd64 the platform root folder will be <root>/linux/amd64 if the folder exists
+//   - Given a target windows/amd64:windows@10.0.20348.1970 the platform root folder will be <root>/windows/amd64/windows@10.0.20348.1970 if the folder exists
+//   - When no target folder exists, the root folder will be equal to <root> folder
+func PlatformRootFolder(root string, target dist.Target, version string) (bool, string) {
+	targets := target.ValuesAsSlice(version)
+	pRootFolder := root
+
+	found := false
+	current := false
+	for _, t := range targets {
+		current, pRootFolder = targetExists(pRootFolder, t)
+		if current {
+			found = current
+		} else {
+			// Not need to keep looking deeply
+			break
+		}
+	}
+	// We will return the last matching folder
+	return found, pRootFolder
+}
+
+func targetExists(root, expected string) (bool, string) {
+	if expected == "" {
+		return false, root
+	}
+	path := filepath.Join(root, expected)
+	if exists, _ := paths.IsDir(path); exists {
+		return true, path
+	}
+	return false, root
+}
+
+func copy(src string, dest string) (string, error) {
```

Comment:
```suggestion
func copyBuildpackTOML(src string, dest string) (string, error) {
```

Reply:
We should probably also be able to handle extension.toml

Reply:
Yeah, I didn't do anything with extensions yet :( 
I will do the following:
- rename this method to `copyFile` adding a new argument `fileName`
- create `copyBuildpackTOML` which invokes `copyFile` passing `buildpack.toml`
- we can add later `copyExtensionTOML`  and reuse the `copyFile` method